### DISCUSSION TITLE
Add support for extending admin api in go plugins

### DISCUSSION THREAD

**Discussion by ealogar** [Kong/kong#7298] [Ideas and feature requests]
### Summary
It would be nice that you can implement an extension to the administration api in go, similarly as you may do with a LUA plugin (e.g. defining an api.go and implementing a custom interface).

**Comment by p0pr0ck5** [comment]
Can I ask, why you would want to do this? Admin API endpoint functionality generally revolves around talking to the datastore, for which there is already extensive functionality in Kong via Lua.

**Comment by ealogar** [comment]
You develop a complex functionality in go that it's not easy to do in LUA, normally there are many libraries in go but LUA lacks of this...
If it were possible to extend admin-api with this code it would bre great... don't you think ?

**Comment by p0pr0ck5** [comment]
Yes I understand what you're saying. :) Can you provide _specific_ examples of where using Go to extend the Admin API would be useful?

**Comment by ealogar** [comment]
I have implemented a go plugin to decrypt JWE tokens. There is not any library in LUA for doing this, but in go you find a go-jose implementation that makes easy to decrypt the tokens.
If I could call this code from an admin-endpoint "/decrypt-token", I could also offer an admin endpoint for inspecting the tokens... as the tokens are encrypted I dont want to offer an api to access them and it's more natural in the admin-api
This is just an example, but you can think in many others.
